Handle missing curl headers

A customer has an issue where they don't have the dev headers for curl installed, so compilation fails.

```
scout_curl_wrapper.c:8:10: fatal error: curl/curl.h: No such file or directory
```

To add headers: `sudo apt-get install libcurl4-openssl-dev`

Beyond the quick fix of adding the header, we need to add support for compiling the rest of the extension without the curl header present. Add a check to the configure script and avoid building the curl instruments if the prereq headers are missing.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -15,7 +15,7 @@ default_job: &default_job
           wget $FINAL_DOWNLOAD_URL
           tar zxf php-$SCOUT_PHP_VERSION.tar.gz
           cd php-$SCOUT_PHP_VERSION
-          ./configure $SCOUT_CONFIGURE_OPTS
+          ./configure $SCOUT_PHP_CONFIGURE_OPTS
           make -j$(nproc)
           sudo make install
           cd ~/repo
@@ -45,105 +45,105 @@ jobs:
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.9
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.1.9-zts":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.9
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.1.32":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.32
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.1.32-zts":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.32
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.1.32-no-curl":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES:
       SCOUT_PHP_VERSION: 7.1.32
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --enable-debug
   "build-7.2.0":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.0
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.2.0-zts":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.0
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.2.24":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.24
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.2.24-zts":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.24
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.2.24-no-curl":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES:
       SCOUT_PHP_VERSION: 7.2.24
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --enable-debug
   "build-7.3.0":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.0
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.3.0-zts":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.0
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.3.11":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.11
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.3.11-zts":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.11
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.3.11-no-curl":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES:
       SCOUT_PHP_VERSION: 7.3.11
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --enable-debug
   "build-7.4.0RC5":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_URL: https://downloads.php.net/~derick/php-7.4.0RC5.tar.gz
       SCOUT_PHP_VERSION: 7.4.0RC5
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.4.0RC5-zts":
     <<: *default_job
     environment:
       ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_URL: https://downloads.php.net/~derick/php-7.4.0RC5.tar.gz
       SCOUT_PHP_VERSION: 7.4.0RC5
-      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+      SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
 
 workflows:
   version: 2

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -8,7 +8,7 @@ default_job: &default_job
     - run:
         name: "Install PHP"
         command: |
-          sudo DEBIAN_FRONTEND=noninteractive apt-get -yq install software-properties-common build-essential autoconf libxml2-dev libsqlite3-dev $ADDITIONAL_SYS_PACKAGES
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -yq install software-properties-common build-essential autoconf $ADDITIONAL_SYS_PACKAGES
           cd /tmp
           [[ -z $SCOUT_PHP_URL ]] && FINAL_DOWNLOAD_URL="https://www.php.net/distributions/php-$SCOUT_PHP_VERSION.tar.gz" || FINAL_DOWNLOAD_URL=$SCOUT_PHP_URL
           echo "Downloading release from $FINAL_DOWNLOAD_URL ..."
@@ -43,104 +43,104 @@ jobs:
   "build-7.1.9":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.9
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.1.9-zts":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.9
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.1.32":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.32
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.1.32-zts":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.32
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.1.32-no-curl":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES:
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev
       SCOUT_PHP_VERSION: 7.1.32
       SCOUT_PHP_CONFIGURE_OPTS: --enable-debug
   "build-7.2.0":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.0
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.2.0-zts":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.0
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.2.24":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.24
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.2.24-zts":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.24
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.2.24-no-curl":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES:
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev
       SCOUT_PHP_VERSION: 7.2.24
       SCOUT_PHP_CONFIGURE_OPTS: --enable-debug
   "build-7.3.0":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.0
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.3.0-zts":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.0
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.3.11":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.11
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.3.11-zts":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.11
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.3.11-no-curl":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES:
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev
       SCOUT_PHP_VERSION: 7.3.11
       SCOUT_PHP_CONFIGURE_OPTS: --enable-debug
   "build-7.4.0RC5":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_URL: https://downloads.php.net/~derick/php-7.4.0RC5.tar.gz
       SCOUT_PHP_VERSION: 7.4.0RC5
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.4.0RC5-zts":
     <<: *default_job
     environment:
-      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
+      ADDITIONAL_SYS_PACKAGES: libxml2-dev libsqlite3-dev libcurl4-openssl-dev
       SCOUT_PHP_URL: https://downloads.php.net/~derick/php-7.4.0RC5.tar.gz
       SCOUT_PHP_VERSION: 7.4.0RC5
       SCOUT_PHP_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -8,14 +8,14 @@ default_job: &default_job
     - run:
         name: "Install PHP"
         command: |
-          sudo DEBIAN_FRONTEND=noninteractive apt-get -yq install software-properties-common build-essential autoconf libcurl4-openssl-dev libxml2-dev libsqlite3-dev
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -yq install software-properties-common build-essential autoconf libxml2-dev libsqlite3-dev $ADDITIONAL_SYS_PACKAGES
           cd /tmp
           [[ -z $SCOUT_PHP_URL ]] && FINAL_DOWNLOAD_URL="https://www.php.net/distributions/php-$SCOUT_PHP_VERSION.tar.gz" || FINAL_DOWNLOAD_URL=$SCOUT_PHP_URL
           echo "Downloading release from $FINAL_DOWNLOAD_URL ..."
           wget $FINAL_DOWNLOAD_URL
           tar zxf php-$SCOUT_PHP_VERSION.tar.gz
           cd php-$SCOUT_PHP_VERSION
-          ./configure --with-curl $SCOUT_CONFIGURE_OPTS
+          ./configure $SCOUT_CONFIGURE_OPTS
           make -j$(nproc)
           sudo make install
           cd ~/repo
@@ -43,75 +43,107 @@ jobs:
   "build-7.1.9":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.9
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.1.9-zts":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.9
-      SCOUT_CONFIGURE_OPTS: --enable-debug --enable-maintainer-zts
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.1.32":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.32
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.1.32-zts":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.1.32
-      SCOUT_CONFIGURE_OPTS: --enable-debug --enable-maintainer-zts
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+  "build-7.1.32-no-curl":
+    <<: *default_job
+    environment:
+      ADDITIONAL_SYS_PACKAGES:
+      SCOUT_PHP_VERSION: 7.1.32
+      SCOUT_CONFIGURE_OPTS: --enable-debug
   "build-7.2.0":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.0
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.2.0-zts":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.0
-      SCOUT_CONFIGURE_OPTS: --enable-debug --enable-maintainer-zts
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.2.24":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.24
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.2.24-zts":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.2.24
-      SCOUT_CONFIGURE_OPTS: --enable-debug --enable-maintainer-zts
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+  "build-7.2.24-no-curl":
+    <<: *default_job
+    environment:
+      ADDITIONAL_SYS_PACKAGES:
+      SCOUT_PHP_VERSION: 7.2.24
+      SCOUT_CONFIGURE_OPTS: --enable-debug
   "build-7.3.0":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.0
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.3.0-zts":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.0
-      SCOUT_CONFIGURE_OPTS: --enable-debug --enable-maintainer-zts
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
   "build-7.3.11":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.11
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.3.11-zts":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_VERSION: 7.3.11
-      SCOUT_CONFIGURE_OPTS: --enable-debug --enable-maintainer-zts
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
+  "build-7.3.11-no-curl":
+    <<: *default_job
+    environment:
+      ADDITIONAL_SYS_PACKAGES:
+      SCOUT_PHP_VERSION: 7.3.11
+      SCOUT_CONFIGURE_OPTS: --enable-debug
   "build-7.4.0RC5":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_URL: https://downloads.php.net/~derick/php-7.4.0RC5.tar.gz
       SCOUT_PHP_VERSION: 7.4.0RC5
-      SCOUT_CONFIGURE_OPTS: --enable-debug
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug
   "build-7.4.0RC5-zts":
     <<: *default_job
     environment:
+      ADDITIONAL_SYS_PACKAGES: libcurl4-openssl-dev
       SCOUT_PHP_URL: https://downloads.php.net/~derick/php-7.4.0RC5.tar.gz
       SCOUT_PHP_VERSION: 7.4.0RC5
-      SCOUT_CONFIGURE_OPTS: --enable-debug --enable-maintainer-zts
+      SCOUT_CONFIGURE_OPTS: --with-curl --enable-debug --enable-maintainer-zts
 
 workflows:
   version: 2
@@ -121,13 +153,16 @@ workflows:
       - "build-7.1.9-zts"
       - "build-7.1.32"
       - "build-7.1.32-zts"
+      - "build-7.1.32-no-curl"
       - "build-7.2.0"
       - "build-7.2.0-zts"
       - "build-7.2.24"
       - "build-7.2.24-zts"
+      - "build-7.2.24-no-curl"
       - "build-7.3.0"
       - "build-7.3.0-zts"
       - "build-7.3.11"
       - "build-7.3.11-zts"
+      - "build-7.3.11-no-curl"
       - "build-7.4.0RC5"
       - "build-7.4.0RC5-zts"

--- a/config.m4
+++ b/config.m4
@@ -14,6 +14,16 @@ PHP_ARG_ENABLE([scoutapm-dev],
 
 if test "$PHP_SCOUTAPM" != "no"; then
 
+  dnl Check if curl is present
+  PHP_CHECK_LIBRARY(curl,curl_easy_perform,
+  [
+    AC_DEFINE(HAVE_SCOUT_CURL,1,[Curl is present on the system])
+  ],[
+    AC_DEFINE(HAVE_SCOUT_CURL,0,[Curl is present on the system])
+  ],[
+    $CURL_LIBS
+  ])
+
   if test "$PHP_SCOUTAPM_DEV" = "yes"; then
     PHP_CHECK_GCC_ARG(-Wbool-conversion,                _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Wbool-conversion")
     PHP_CHECK_GCC_ARG(-Wdeclaration-after-statement,    _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Wdeclaration-after-statement")

--- a/config.m4
+++ b/config.m4
@@ -20,6 +20,7 @@ if test "$PHP_SCOUTAPM" != "no"; then
     AC_DEFINE(HAVE_SCOUT_CURL,1,[Curl is present on the system])
   ],[
     AC_DEFINE(HAVE_SCOUT_CURL,0,[Curl is present on the system])
+    AC_MSG_WARN([curl library headers were not found on the system, scoutapm will not instrument curl functions])
   ],[
     $CURL_LIBS
   ])

--- a/package.xml
+++ b/package.xml
@@ -25,11 +25,11 @@
     </lead>
 
     <!-- Current Release -->
-    <date>2019-11-06</date>
-    <time>08:26:00</time>
+    <date></date>
+    <time></time>
     <version>
-        <release>1.0.1</release>
-        <api>1.0.1</api>
+        <release></release>
+        <api></api>
     </version>
     <stability>
         <release>stable</release>
@@ -37,9 +37,7 @@
     </stability>
     <license uri="https://opensource.org/licenses/MIT">MIT</license>
     <notes>
-        - Fix segfault when trying to access args out of bounds (#48)
-        - Fix exception raised when trying to fopen a file that does not exist (#50)
-        - Removed notice emitted calling some functions (#51)
+        - Check if curl header is available and compile support with/without curl accordingly (#54)
     </notes>
     <!-- End Current Release -->
 
@@ -94,6 +92,24 @@
     <zendextsrcrelease />
 
     <changelog>
+        <release>
+            <date>2019-11-06</date>
+            <time>08:26:00</time>
+            <version>
+                <release>1.0.1</release>
+                <api>1.0.1</api>
+            </version>
+            <stability>
+                <release>stable</release>
+                <api>stable</api>
+            </stability>
+            <license uri="https://opensource.org/licenses/MIT">MIT</license>
+            <notes>
+                - Fix segfault when trying to access args out of bounds (#48)
+                - Fix exception raised when trying to fopen a file that does not exist (#50)
+                - Removed notice emitted calling some functions (#51)
+            </notes>
+        </release>
         <release>
             <date>2019-11-04</date>
             <time>11:50:00</time>

--- a/scout_curl_wrapper.c
+++ b/scout_curl_wrapper.c
@@ -5,8 +5,10 @@
  * For license information, please see the LICENSE file.
  */
 
-#include <curl/curl.h>
 #include "zend_scoutapm.h"
+
+#if HAVE_SCOUT_CURL
+#include <curl/curl.h>
 #include "scout_extern.h"
 
 ZEND_NAMED_FUNCTION(scoutapm_curl_setopt_handler)
@@ -60,3 +62,4 @@ ZEND_NAMED_FUNCTION(scoutapm_curl_exec_handler)
         SCOUTAPM_G(disconnected_call_argument_store)[recorded_arguments_index].argv
     );
 }
+#endif

--- a/tests/009-curl_exec.phpt
+++ b/tests/009-curl_exec.phpt
@@ -3,6 +3,7 @@ Calls to curl_exec are logged
 --SKIPIF--
 <?php if (!extension_loaded("scoutapm")) die("skip scoutapm extension required."); ?>
 <?php if (!extension_loaded("curl")) die("skip curl extension required."); ?>
+<?php if (!in_array("curl_exec", scoutapm_list_instrumented_functions())) die("skip scout was not compiled with curl");
 --FILE--
 <?php
 $ch = curl_init();

--- a/zend_scoutapm.c
+++ b/zend_scoutapm.c
@@ -55,6 +55,7 @@ ZEND_DECLARE_MODULE_GLOBALS(scoutapm)
 /* a PHP module defines what functions it exports */
 static const zend_function_entry scoutapm_functions[] = {
     PHP_FE(scoutapm_get_calls, NULL)
+    PHP_FE(scoutapm_list_instrumented_functions, NULL)
     PHP_FE_END
 };
 
@@ -487,4 +488,27 @@ PHP_FUNCTION(scoutapm_get_calls)
 
     SCOUTAPM_DEBUG_MESSAGE("done.\n");
 }
+
+
+/* {{{ proto array scoutapm_list_instrumented_functions()
+   Fetch a list of functions that will be instrumented or monitored by the ScoutAPM extension. */
+PHP_FUNCTION(scoutapm_list_instrumented_functions)
+{
+    int i, lookup_count = sizeof(handler_lookup) / sizeof(indexed_handler_lookup);
+
+    array_init(return_value);
+
+    for(i = 0; i < lookup_count; i++) {
+        if (original_handlers[handler_lookup[i].index] == NULL) {
+            continue;
+        }
+
+        add_next_index_stringl(
+            return_value,
+            handler_lookup[i].function_name,
+            strlen(handler_lookup[i].function_name)
+        );
+    }
+}
+
 /* }}} */

--- a/zend_scoutapm.c
+++ b/zend_scoutapm.c
@@ -20,8 +20,10 @@ static int zend_scoutapm_startup(zend_extension*);
 static void free_recorded_call_arguments();
 static int unchecked_handler_index_for_function(const char *function_to_lookup);
 
+#if HAVE_SCOUT_CURL
 extern ZEND_NAMED_FUNCTION(scoutapm_curl_setopt_handler);
 extern ZEND_NAMED_FUNCTION(scoutapm_curl_exec_handler);
+#endif
 extern ZEND_NAMED_FUNCTION(scoutapm_fopen_handler);
 extern ZEND_NAMED_FUNCTION(scoutapm_fread_handler);
 extern ZEND_NAMED_FUNCTION(scoutapm_fwrite_handler);
@@ -165,8 +167,10 @@ static PHP_RINIT_FUNCTION(scoutapm)
         /* @todo make overloaded functions configurable? https://github.com/scoutapp/scout-apm-php-ext/issues/30 */
         SCOUT_OVERLOAD_FUNCTION("file_get_contents", scoutapm_default_handler)
         SCOUT_OVERLOAD_FUNCTION("file_put_contents", scoutapm_default_handler)
+#if HAVE_SCOUT_CURL
         SCOUT_OVERLOAD_FUNCTION("curl_setopt", scoutapm_curl_setopt_handler)
         SCOUT_OVERLOAD_FUNCTION("curl_exec", scoutapm_curl_exec_handler)
+#endif
         SCOUT_OVERLOAD_FUNCTION("fopen", scoutapm_fopen_handler)
         SCOUT_OVERLOAD_FUNCTION("fwrite", scoutapm_fwrite_handler)
         SCOUT_OVERLOAD_FUNCTION("fread", scoutapm_fread_handler)

--- a/zend_scoutapm.h
+++ b/zend_scoutapm.h
@@ -23,6 +23,7 @@
 #define SCOUT_APM_EXT_DEBUGGING 0
 
 PHP_FUNCTION(scoutapm_get_calls);
+PHP_FUNCTION(scoutapm_list_instrumented_functions);
 
 /* Describes information we store about a recorded stack frame */
 typedef struct _scoutapm_stack_frame {

--- a/zend_scoutapm.h
+++ b/zend_scoutapm.h
@@ -8,9 +8,7 @@
 #ifndef ZEND_SCOUTAPM_H
 #define ZEND_SCOUTAPM_H
 
-#ifdef HAVE_CONFIG_H
 #include "config.h"
-#endif
 
 #include "php.h"
 #include <zend_extensions.h>

